flink-python/pyflink/serializers.py

```diff
@@ -0,0 +1,192 @@
+################################################################################
+#  Licensed to the Apache Software Foundation (ASF) under one
+#  or more contributor license agreements.  See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+import sys
+import struct
+
+if sys.version < '3':
+    import cPickle as pickle
+    protocol = 2
+    from itertools import imap as map
+else:
+    import pickle
+    protocol = 3
+    xrange = range
+
+
+class SpecialLengths(object):
+    END_OF_DATA_SECTION = -1
+
+
+class Serializer(object):
+
+    def dump_stream(self, iterator, stream):
+        """
+        Serializes an iterator of objects to the output stream.
+        """
+        raise NotImplementedError
```

Comment:
How about using `abc.ABCMeta` to mark the abstract method instead of `NotImplementedError `? I think using `abc.ABCMeta` is more semantically clear.